fed/mosaic-sumo/src/main/java/org/eclipse/mosaic/fed/sumo/ambassador/SumoAmbassador.java

```diff
@@ -331,37 +354,33 @@ protected void flushNotYetAddedVehicles(long time) throws InternalFederateExcept
     }
 
     private void applyChangesInVehicleTypeForVehicle(String vehicleId, VehicleType actualVehicleType, VehicleType baseVehicleType) throws InternalFederateException {
-        if (!eq(actualVehicleType.getTau(), baseVehicleType.getTau())) {
+        double epsilon = 1e-4;
+        if (!MathUtils.isFuzzyEqual(actualVehicleType.getTau(), baseVehicleType.getTau(), epsilon)) {
             double minReactionTime = sumoConfig.updateInterval / 1000d;
             traci.getVehicleControl().setReactionTime(
                     vehicleId, Math.max(minReactionTime, actualVehicleType.getTau() + sumoConfig.timeGapOffset)
             );
         }
-        if (!eq(actualVehicleType.getMaxSpeed(), baseVehicleType.getMaxSpeed())) {
+        if (!MathUtils.isFuzzyEqual(actualVehicleType.getMaxSpeed(), baseVehicleType.getMaxSpeed(), epsilon)) {
             traci.getVehicleControl().setMaxSpeed(vehicleId, actualVehicleType.getMaxSpeed());
         }
-        if (!eq(actualVehicleType.getAccel(), baseVehicleType.getAccel())) {
+        if (!MathUtils.isFuzzyEqual(actualVehicleType.getAccel(), baseVehicleType.getAccel(), epsilon)) {
             traci.getVehicleControl().setMaxAcceleration(vehicleId, actualVehicleType.getAccel());
         }
-        if (!eq(actualVehicleType.getDecel(), baseVehicleType.getDecel())) {
+        if (!MathUtils.isFuzzyEqual(actualVehicleType.getDecel(), baseVehicleType.getDecel(), epsilon)) {
```

Comment:
do we need our own epsilon here? Have you tried if it works with `isFuzzyEqual(a,b)`?

Reply:
No, I haven't tried, I just assumed the author chose the used value for a reason.

I still consider this a useful addition though. But I also understand if we want to keep the code base "slim".
I will check if the default epsilon works for all tests.